src/helpers/slack.ts

```diff
@@ -10,29 +11,59 @@ import {
 const fs = require("fs");
 
 export function getEmojisToReactWith(text: string): Array<string> {
-    const emojisToReactWith: Array<string> = [];
+    const lowerCaseText = text.toLowerCase();
 
     // Read list of emojis from file
-    const rawdata = fs.readFileSync(
-        process.env.EMOJIS_FILE_PATH || "./helpers/files/emojis.json"
-    );
-    const emojis: Record<string, Array<string>> = JSON.parse(rawdata);
+    const emojis: Record<string, Array<string>> = readEmojisFromFile();
 
-    // flattens the array of keywords
     const keywords = Object.values(emojis).flat();
 
     // search for each keyword in the text
+    const emojisToReactWith = findMatchingEmojiKeywords(
+        keywords,
+        lowerCaseText,
+        emojis
+    );
+
+    return Array.from(new Set(emojisToReactWith));
+}
+
+function readEmojisFromFile(): Record<string, Array<string>> {
+    var emojisFilePath: string;
+    if (process.env.EMOJIS_FILE_PATH) {
+        // check if the file path is valid
+        if (!fs.existsSync(process.env.EMOJIS_FILE_PATH)) {
```

Comment:
[Non blocking comment]

Whenever config file path needs to be given to a program, it is good to check three things, PATH, NAME and FILE TYPE to ensure correct file was provided. What if correct path was provided but wrong file say `/home/demo.txt`, this will set the value of `emojisFilePath` but yield in incorrect result.

Reply:
The `try..catch` block will catch any failures when parsing JSON data from the file and we check if file exists, let me know if this is what you were expecting?

Reply:
`try..catch` is used to handle errors gracefully, but here if something like `/home/demo.txt` is passed, it will not throw an error. You might need to be specific about checking

Reply:
Sorry it took this long, I have added a check to see if the parsed JSON data is in the format `Record<string, Array<string>>`

Do you recommend any other checks?

Reply:
you can just look into file before parsing eg: filename, file extension, etc. If they are not valid, do not parse at all as checking file type is looking ahead rather than parsing the whole doc and then realizing it not the correct one.

this should be good for now.